### PR TITLE
docs(node): coordinator joining ceremony

### PR DESCRIPTION
**Pull request by fbozic**
- Document basic node struct.
- Create placeholders for the things that need to be documented.
- Document coordinator joining ceremony. 

Resolves: #582

### CODE REVIEW

File: crates/node/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+### Coordinator joining ceremony
+
+Coorindator joining ceremony is the process of acquiring the coordinator node for the context.
+
+Setup:
+- Coordinators and peers are subscribed to the dedicated coordinator ceremony topic.
+- Context creator sends a message to the topic requesting a coordinator.
+- Interested coordinators send a message to the topic replying with the intent to coordinate.
+- Context creator selects a coordinator from the interested coordinators.
```
Comment (MatejVukosav) [inline crates/node/README.md:62]: How will this look like?

Comment (fbozic) [reply crates/node/README.md:62]: Initial implementation is accept the first one, and future implementation can be based on staked amount or similar.

**Review by MatejVukosav — APPROVED** [review]

File: crates/node/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+TODO: Write about the transaction handling process and draw sequence diagram
+
+### Coordinator joining ceremony
+
+Coorindator joining ceremony is the process of acquiring the coordinator node for the context.
+
+Setup:
+- Coordinators and peers are subscribed to the dedicated coordinator ceremony topic.
+- Context creator sends a message to the topic requesting a coordinator.
```
Comment (miraclx) [inline crates/node/README.md:60]: is this at the point of context creation? otherwise will this require the context creator to always be online?

**Review by miraclx — APPROVED** [review]
lgtm, now we can deny random confirmations from other nodes attempting to take over the network

File: crates/node/README.md (hunk 2 above)
Comment (fbozic) [reply crates/node/README.md:60]: good question, I would say yes although I see the issues it brings.

do you maybe have some alternative approach?

Comment (miraclx) [reply crates/node/README.md:60]: At the point of creation is fine, so everyone expects to use the same coordinator. And if they're offline, nothing happens.

So far as catchup sends the coordinator all the transactions in the pool when they come back online. We need to verify that's the case.

Comment (fbozic) [reply crates/node/README.md:60]: I've linked this comment to an implementation issue so we don't forget about it.

**Pull request merged**